vector/src/main/java/im/vector/app/core/ui/views/TypingMessageAvatar.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.ui.views
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.LinearLayout
+import im.vector.app.core.utils.DimensionConverter
+import im.vector.app.features.home.AvatarRenderer
+import org.matrix.android.sdk.api.session.room.sender.SenderInfo
+import org.matrix.android.sdk.api.util.toMatrixItem
+
+class TypingMessageAvatar @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    companion object {
+        const val AVATAR_SIZE_DP = 20
+        const val OVERLAP_FACT0R = -3 // =~ 30% to left
+    }
+
+    fun render(typingUsers: List<SenderInfo>, avatarRender: AvatarRenderer) {
```

Comment:
have you noticed any performance issues with creating/dropping views as part of the typing flow?

I'm wondering if it's worth caching and reusing the imageviews instead of dynamically creating/removing them, probably not~

Reply:
I do not think we need to handle cache at this level. 
avatars count are limit to 4, and users could change rapidly in a short time. 
in total the view took almost 1 second and disappear. Still I will try to renforce test on this. 
Thank you